types/module/configurator.go

```diff
@@ -15,16 +19,35 @@ type Configurator interface {
 	// QueryServer returns a grpc.Server instance which allows registering services
 	// that will be exposed as gRPC services as well as ABCI query handlers.
 	QueryServer() grpc.Server
+
+	// RegisterMigration registers an in-place store migration for a module. The
+	// handler is a migration script to perform in-place migrations from version
+	// `fromVersion` to version `fromVersion+1`.
+	RegisterMigration(moduleName string, fromVersion uint64, handler func(store sdk.KVStore) error) error
```

Comment:
Should we use KVStore or sdk.Context?

Reply:
If sdk.Context, then in [x/bank's migration function](https://github.com/cosmos/cosmos-sdk/blob/b7141f990988ea994788c1e350e6b334004bd868/x/bank/legacy/v042/store.go#L38), I would need to do: `store := ctx.KVStore(bankStoreKey)`, so I would need `bankStoreKey` there.

The function sig here is either `func(store sdk.KVStore)` or `func(ctx sdk.Context, storeKey sdk.StoreKey)`, i thought the 1st one was clearer.

Reply:
`sdk.Context` includes block header information which could be useful for migrations, no? I would say `sdk.Context` is more consistent with other handler usage and more future proof.